### PR TITLE
Implement new wasp/server/email API

### PR DESCRIPTION
**Pull request by infomiho**
Moved the whole `sdk/email` folder to `sdk/server/email`.

### CODE REVIEW

File: waspc/data/Generator/templates/server/src/crud/_operations.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ import type {
 {=# isAuthEnabled =}
 import { throwInvalidCredentialsError } from 'wasp/auth/utils'
 {=/ isAuthEnabled =}
-import type { GetAllQuery, GetQuery, CreateAction, UpdateAction, DeleteAction } from "{= crudTypesImportPath =}";
+import type { {= crud.name =} } from "wasp/server/crud";
```
Comment (infomiho) [inline waspc/data/Generator/templates/server/src/crud/_operations.ts:11]: Fixing the CRUD bug in `new-sdk` branch (should be resolved after we merge the `wasp/client/crud` API changes)

File: waspc/data/Generator/templates/sdk/package.json [hunk 2]
```diff
@@ -89,9 +89,8 @@
       "./crud/*": "./dist/crud/*.js",
       {=!  Used by our code, uncodumented (but accessible) for users. =}
       "./server/crud/*": "./dist/server/crud/*",
-      "./email": "./dist/email/index.js",
       {=!  Used by our code, uncodumented (but accessible) for users. =}
-      "./email/core/types": "./dist/email/core/types.js",
+      "./server/email/core/types": "./dist/server/email/core/types.js",
```
Comment (Martinsos) [inline waspc/data/Generator/templates/sdk/package.json:93]: If you know we still need these, it is maybe best if you move it below to section "still needed, reconsider during refactoring."? No need to do it now, just idea for future PRs.

File: waspc/src/Wasp/Generator/SdkGenerator/Server/EmailSenderG.hs [hunk 3]
```diff
@@ -17,8 +17,8 @@ import qualified Wasp.Generator.SdkGenerator.Common as C
 import qualified Wasp.Generator.SdkGenerator.EmailSender.Providers as Providers
 import Wasp.Util ((<++>))
 
-genEmailSender :: AppSpec -> Generator [FileDraft]
-genEmailSender spec = case maybeEmailSender of
+genNewEmailSenderApi :: AppSpec -> Generator [FileDraft]
```
Comment (Martinsos) [inline waspc/src/Wasp/Generator/SdkGenerator/Server/EmailSenderG.hs:20]: Why `new`?

**Review by Martinsos — APPROVED** [review]
LGTM but take a look at the comments.

**Pull request merged**